### ISSUE TITLE
Right-clicking part builder crashes client while in spectator

### ISSUE THREAD

**Issue by UnRealDinnerbone** [SlimeKnights/TinkersConstruct#5005]
### Minecraft Version

1.18.2

### Forge Version

40.1.73

### Mantle Version

1-18.2-1.9.27

### Tinkers' Construct Version

1.18.2-3.5.2.40

### Describe your issue

Crash when right-clicking the Part Builder while in spectator mode 

### Crash Report

https://pste.ch/acuzujuvon.hs

### Other mods

Just Tinkers

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

None of the above

### Searched for known issues?

Searched open issues, Searched closed issues

**Comment by ShinyPorygon** [comment]
This is not a crash log that is just tinkers. If you reproduced with only the tinkers + mantle mod we need that crash log. If you didn't do so, don't say you did.

**Comment by UnRealDinnerbone** [comment]
Updated Crashlog: https://pste.ch/jabozejoku.hs

**Comment by KnightMiner** [comment]
I pushed some code which is a blanket fix for all our UI blocks to make them visible in spectator mode. The blanket fix does make the assumption that the block actually has a UI, so its possible somewhere I forgot to mark it has having no UI and it will cause a crash. Its also possible spectators can interact with some UI they shouldn't be able to

I tested in the various key UIs such as the smeltery and tables, and they seemed to work fine so hopefully there are no issues.
